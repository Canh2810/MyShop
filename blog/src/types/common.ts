export enum TypoVariants {
  TextExtraSmall = 'textExtraSmall',
  TextSmall = 'textSmall',
  TextMedium = 'textMedium',
  TextLarge = 'textLarge',
  TexExtraLarge = 'textExtraLarge',
  HeadingSmall = 'headingSmall',
  HeadingMedium = 'headingMedium',
  HeadingLarge = 'headingLarge',
  HeadingExtraLarge = 'headingExtraLarge',
}

export enum TypoColors {
  Primary = 'textPrimary',
  Secondary = 'textSecondary',
  Tertiary = 'textTertiary',
}

export enum ButtonVariants {
  Container = 'container',
  Outlined = 'outlined',
}

export enum CommonVariants {
  Primary = 'primary',
  Secondary = 'secondary',
}

export enum TextFieldVariants {
  Outlined = 'outlined',
  Filled = 'filled',
}

export enum TextFieldTypes {
  Text = 'text',
  Password = 'password',
}

export enum MenuVariants {
  Vertical = 'vertical',
  Horizontal = 'horizontal',
}

export interface IMenuItem {
  title: string
  href: string
}
