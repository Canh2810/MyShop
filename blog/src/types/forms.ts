export interface ILoginForm {
  email: string
  password: string
}

export type LoginFormField = Exclude<keyof ILoginForm, ''>
