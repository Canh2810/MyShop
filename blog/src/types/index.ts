export * from './common'
export * from './posts'
export * from './forms'
export * from './users'
export * from './response'
