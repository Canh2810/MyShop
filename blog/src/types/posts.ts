import { FavoriteStatus } from '.'

export interface IPost {
  id: string
  category: string
  title: string
  author: {
    avatar: string
    username: string
  }
  date: string
  imageURL: string
  favorite: FavoriteStatus
}
