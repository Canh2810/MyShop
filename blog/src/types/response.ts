import { User, Status } from '.'

export interface LoginResponse {
  status: Status
  message?: string
  data?: User
}
