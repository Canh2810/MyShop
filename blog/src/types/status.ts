export enum CommonStatus {
  Failed,
  Success,
}

export enum FavoriteStatus {
  NotFavorite,
  Favorite,
}
