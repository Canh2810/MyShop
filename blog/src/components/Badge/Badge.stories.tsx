import { StoryObj, Meta } from '@storybook/react'

import Badge from '.'
import { CommonVariants } from '@/types'

export default {
  title: 'Components/Badge',
  component: Badge,
} as Meta

type Story = StoryObj<typeof Badge>

export const Default: Story = {
  args: {
    title: 'Badge',
  },
}

export const Secondary: Story = {
  args: {
    title: 'Badge',
    variant: CommonVariants.Secondary,
  },
}
