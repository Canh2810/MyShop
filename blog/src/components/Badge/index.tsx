// Types
import { CommonVariants } from '@/types'

export interface BadgeProps {
  variant?: CommonVariants
  title: string
}

const Badge = ({ variant = CommonVariants.Primary, title }: BadgeProps) => {
  const badgeStyles = {
    [CommonVariants.Primary]: 'bg-primary-500 text-white',
    [CommonVariants.Secondary]: 'bg-dark-800 text-primary-500',
  }

  return (
    <span
      className={`${badgeStyles[variant]} rounded-md px-2.5 py-1 text-xs font-normal leading-3`}
    >
      {title}
    </span>
  )
}

export default Badge
