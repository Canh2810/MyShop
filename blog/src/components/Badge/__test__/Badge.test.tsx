import { render } from '@testing-library/react'
import Badge from '..'
import { CommonVariants } from '@/types'

describe('Badge component', () => {
  it('should render primary variant', () => {
    const { container } = render(<Badge title="Test Badge" />)
    expect(container).toMatchSnapshot()
  })
  it('should render secondary variant', () => {
    const { container } = render(
      <Badge variant={CommonVariants.Secondary} title="Test Badge" />,
    )
    expect(container).toMatchSnapshot()
  })
})
