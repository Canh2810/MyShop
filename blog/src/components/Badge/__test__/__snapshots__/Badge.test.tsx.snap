// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge component should render primary variant 1`] = `
<div>
  <span
    class="bg-primary-500 text-white rounded-md px-2.5 py-1 text-xs font-normal leading-3"
  >
    Test Badge
  </span>
</div>
`;

exports[`Badge component should render secondary variant 1`] = `
<div>
  <span
    class="bg-dark-800 text-primary-500 rounded-md px-2.5 py-1 text-xs font-normal leading-3"
  >
    Test Badge
  </span>
</div>
`;
